### ISSUE TITLE
Integration test suite

### ISSUE THREAD

**Issue by lazypower** [juju-solutions/charms.docker#34]
Per a conversation with @simonklb, our unit test suite is great... if you only care about the python wrapping code. There's currently no integration tests being run so its entirely possible we will break on a docker update that changes the CLI command structure.

We can reasonably test that the charms.docker interface is still valid with some rudimentary integration  tests running in travis using DIND

Basically go through all of the docker and docker-compose commands on a real docker container and docker compose manifest would be great
